### ISSUE TITLE
twisted.web.http.Request.addCookie does not support SameSite=None

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10088]
|[<img alt="richvdh's avatar" src="https://avatars.githubusercontent.com/u/1389908?s=50" width="50" height="50">](https://github.com/richvdh)| @richvdh reported|
|-|-|
|Trac ID|trac#10088|
|Type|defect|
|Created|2021-02-10 13:12:26Z|

It looks like the support for `SameSite` on `Set-Cookie` headers, which was added by https://github.com/twisted/twisted/pull/997, was based on a draft of the RFC, https://tools.ietf.org/html/draft-west-first-party-cookies-07, and only accepts the values `lax` or `strict`.

This seems to have been superceded by https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-07, which adds a `None` option, and, as described on https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite, most browsers now default to `Lax` rather than `None`, so it's necessary to explicitly set it to `None` if you want that behaviour.

In short, it would be good to allow `"none"` here.

<details><summary>Searchable metadata</summary>

```
trac-id__10088 10088
type__defect defect
reporter__richvdh richvdh
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__new new
resolution__None None
component__web web
keywords__None None
time__1612962746474365 1612962746474365
changetime__1620472010362606 1620472010362606
version__None None
owner__None None

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

For reference. The initial `sameSite` was implemented in [#9387](https://github.com/twisted/twisted/issues/9387)

Happy to review and accept a PR for `sameSite=None` :)

**Comment by 5er9e1** [comment]
@adiroiban 
Added support for `SameSite=None` [here](https://github.com/twisted/twisted/pull/12381).

**Comment by 5er9e1** [comment]
@adiroiban
Sorry for this big delay. I was busy with my main projects. What is about PR10088? If I understand correctly the last part that we [have to fix is a API Docs](https://github.com/twisted/twisted/pull/12381#discussion_r1874406849) that should be updated?

I'm not too much familiar how to update documentation? Could you help me with it? Also I'm not sure how is working workflow since PR and before merging, but if I understand correctly, the updating of docs is a last thing that I have to do (I can't see any buttons to be able merge), and after this I just need to wait until it will be merged?
